app-typescript/components/ToggleBox/CustomHeaderToggleBox.tsx

```diff
@@ -22,10 +26,31 @@ export class CustomHeaderToggleBox extends React.PureComponent<IPropsCustomHeade
         });
     }
 
+    componentDidUpdate(_prevProps: IPropsCustomHeader, prevState: IState) {
+        if (prevState.isOpen !== this.state.isOpen) {
+            this.setState({ isAnimating: true });
+
+            if (this.contentRef.current) {
+                this.contentRef.current.addEventListener('animationend', this.handleAnimationEnd);
+            }
+        }
+    }
+
+    handleAnimationEnd = () => {
+        this.setState({ isAnimating: false });
+
+        if (this.contentRef.current) {
+            this.contentRef.current.removeEventListener('animationend', this.handleAnimationEnd);
+        }
+    }
+
     render() {
-        let classes = classNames('sd-shadow--z1 new-collapse-box', {
+        const classes = classNames('sd-shadow--z1 new-collapse-box', {
             'new-collapse-box--open': this.state.isOpen,
         });
+        const childrenClasses = classNames('new-collapse-box__content-inner p-2 pt-0-5', {
```

Comment:
`childrenClasses` doesn't sound like a great name. How about inlining the code so there's no confusion where it is applied?